lib/samples/show_grid/show_grid_sample.dart

```diff
@@ -0,0 +1,368 @@
+  void _onLabelVisibilityChanged(bool value) {
+    if (_mapViewController.grid != null) {
+      _mapViewController.grid!.labelVisibility = value;
+    }
+  }
+
+  // change the label color based on the given value.
+  void _onLabelColorTypeChanged(Color color) {
+    if (_mapViewController.grid != null) {
+      final grid = _mapViewController.grid!;
+      for (int i = 0; i < grid.levelCount; i++) {
+        final textSymbol = TextSymbol(
+          color: color,
+          size: 14.0,
+          horizontalAlignment: HorizontalAlignment.left,
+          verticalAlignment: VerticalAlignment.bottom,
+        )
+          ..haloColor = Colors.black
+          ..haloWidth = 5.0;
+        grid.setTextSymbol(level: i, textSymbol: textSymbol);
+      }
+    }
+  }
+
+  // change the label position based on the given value.
+  void _onLabelPositionTypeChanged(GridLabelPosition labelPosition) {
+    if (_mapViewController.grid != null) {
+      final grid = _mapViewController.grid!;
+      grid.labelPosition = labelPosition;
+    }
+  }
+
+  /// Show the grid options dialog.
+  void _showGridOptions() async {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) {
+        return AlertDialog(
+          title: const Text('Grid Options'),
+          content: GridOptions(
+            onGridTypeChanged: _onGridTypeChanged,
+            onGridColorTypeChanged: _onGridColorTypeChanged,
+            onLabelColorTypeChanged: _onLabelColorTypeChanged,
+            onLabelPositionTypeChanged: _onLabelPositionTypeChanged,
+            onLabelVisibilityChanged: _onLabelVisibilityChanged,
+          ),
+          actions: <Widget>[
+            TextButton(
+              child: const Text('Close'),
+              onPressed: () {
+                Navigator.of(context).pop();
+              },
+            ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+///
+/// a widget that displays grid options.
+///
+class GridOptions extends StatefulWidget {
+  final Function(Grid) onGridTypeChanged;
+  final Function(Color) onGridColorTypeChanged;
+  final Function(Color) onLabelColorTypeChanged;
+  final Function(GridLabelPosition) onLabelPositionTypeChanged;
+  final Function(bool) onLabelVisibilityChanged;
+
+  GridOptions({
+    Key? key,
+    required this.onGridTypeChanged,
+    required this.onGridColorTypeChanged,
+    required this.onLabelColorTypeChanged,
+    required this.onLabelPositionTypeChanged,
+    required this.onLabelVisibilityChanged,
+  }) : super(key: key);
+
+  @override
+  State<GridOptions> createState() => _GridOptionsState();
+}
+
+class _GridOptionsState extends State<GridOptions> {
+  final gridTypes = ['Latitude & Longitude', 'MGRS', 'UTM', 'USNG'];
+  final colorTypes = ['Red', 'Blue', 'Green', 'Yellow'];
+  final labelPositionTypes = [
+    'AllSides',
+    'BottomLeft',
+    'BottomRight',
+    'Center',
+    'Geographic',
+    'TopLeft',
+    'TopRight'
+  ];
```

Comment:
What would you think about making these enums? That would simplify some things, but make other things a little more complex. I can't decide if that would be better.

Reply:
I suppose it's nice that they can all use `_createDropdownButtonFormField()`.

Reply:
I keep them in the string list to simplify implementation and reuse the creat function.

Reply:
Ok, that makes sense.